src/api/layer.js

```diff
@@ -157,16 +148,11 @@ export default class Layer {
      */
     setStyle(style) {
         this._checkStyle(style);
-        return this._styleChanged(style).then(r => {
-            if (this._style) {
-                this._style.onChange(null);
-            }
-            this._style = style;
-            this._style.onChange(() => {
-                this._styleChanged(style);
-            });
-            return r;
-        });
+        this._style.onChange(null);
+        this._style = style;
```

Comment:
Similar problem here, this sets the style without checking that the style is valid for the source